lib/BaseClient.ts

```diff
@@ -45,28 +45,47 @@ abstract class BaseClient extends EventEmitter {
    * Returns the total balance available across all channels.
    */
   public abstract channelBalance(): Promise<ChannelBalance>;
-  protected setStatus(status: ClientStatus): void {
+
+  protected setStatus = async (status: ClientStatus): Promise<void> => {
     this.logger.info(`${this.constructor.name} status: ${ClientStatus[status]}`);
     this.status = status;
-    this.checkTimers();
+    await this.setTimers();
   }
-  private checkTimers() {
+
+  private setTimers = async () => {
     if (this.status === ClientStatus.ConnectionVerified) {
-      this.updateCapacityTimer = setInterval(async () => {
-        try {
-          this.maximumOutboundCapacity = (await this.channelBalance()).balance;
-        } catch (e) {
-          // TODO: Mark client as disconnected
-          this.logger.error(`failed to fetch channelbalance from client: ${e}`);
-        }
-      }, this.CAPACITY_REFRESH_INTERVAL);
+      await this.updateCapacity();
+      this.updateCapacityTimer = setInterval(this.updateCapacity, BaseClient.CAPACITY_REFRESH_INTERVAL);
+
+      if (this.reconnectionTimer) {
+        clearTimeout(this.reconnectionTimer);
+        this.reconnectionTimer = undefined;
+      }
     } else {
       if (this.updateCapacityTimer) {
         clearInterval(this.updateCapacityTimer);
       }
+      if (this.status !== ClientStatus.Disabled) {
```

Comment:
I believe this change is causing a memory leak and preventing my xud from shutting down properly when using `npm run dev:watch`:

```
[1] 19/04/2019 12:35:16.611 [LND] info: LndClient status: Disconnected
[1] 19/04/2019 12:35:16.611 [LND] error: could not verify connection to lnd for BTC at localhost:10009, error: {"code":14,"metadata":{"_internal_repr":{}},"details":"Connect Failed"},
[1]           retrying in 5000 ms
[1] 19/04/2019 12:35:16.612 [LND] info: LndClient status: Disconnected
[1] 19/04/2019 12:35:21.614 [LND] info: trying to verify connection to lnd for BTC at localhost:10009
[1] 19/04/2019 12:35:21.615 [LND] info: trying to verify connection to lnd for BTC at localhost:10009
[1] 19/04/2019 12:35:27.408 [LND] error: could not verify connection to lnd for BTC at localhost:10009, error: {"code":14,"metadata":{"_internal_repr":{}},"details":"Connect Failed"},
[1]           retrying in 5000 ms
[1] 19/04/2019 12:35:27.409 [LND] info: LndClient status: Disconnected
[1] 19/04/2019 12:35:27.409 [LND] error: could not verify connection to lnd for BTC at localhost:10009, error: {"code":14,"metadata":{"_internal_repr":{}},"details":"Connect Failed"},
[1]           retrying in 5000 ms
[1] 19/04/2019 12:35:27.409 [LND] info: LndClient status: Disconnected
^C[1] 19/04/2019 12:35:28.406 [GLOBAL] info: XUD is shutting down
[1] 19/04/2019 12:35:28.408 [RPC] info: GRPC server completed shutdown
[1] 19/04/2019 12:35:28.409 [GLOBAL] info: XUD shutdown gracefully
[0] npm run compile:watch exited with code 0
--> Sending SIGTERM to other processes..
[1] 19/04/2019 12:35:32.411 [LND] info: trying to verify connection to lnd for BTC at localhost:10009
[1] 19/04/2019 12:35:39.091 [LND] error: could not verify connection to lnd for BTC at localhost:10009, error: {"code":14,"metadata":{"_internal_repr":{}},"details":"Connect Failed"},
[1]           retrying in 5000 ms
[1] 19/04/2019 12:35:39.091 [LND] info: LndClient status: Disconnected
^C[1] 19/04/2019 12:35:41.470 [GLOBAL] info: XUD is already shutting down
^C
[1] 19/04/2019 12:35:42.220 [GLOBAL] info: XUD is already shutting down
➜  xud git:(fix/capacity-checks) ✗ pgrep node | xargs kill
[1] 19/04/2019 12:35:44.096 [LND] info: trying to verify connection to lnd for BTC at localhost:10009
[1] npm run nodemon:watch exited with code 0
```

Steps to replicate:
1. start xud with lnd enabled, but stopped/disconnected
2. let it try reconnect for 2+ cycles
3. start lnd
4. close `npm run dev:watch` with ctrl+c
5. ERROR - the process is not being killed properly (I'm unable to reproduce the same behaviour in current master)

Reply:
```
      if (this.status !== ClientStatus.Disabled) {
        /*
        if (this.reconnectionTimer) {
          console.log('debug(karl): reconnectionTimer already exists -> overwriting this.reconnectionTimer with new one');
        }
        */
        this.reconnectionTimer = setTimeout(this.verifyConnection, BaseClient.RECONNECT_TIMER);
      }
```

Reply:
Is the issue with overwriting the timer without clearing it, you think? That would make sense to me, I just didn't realize we were setting it to disconnected twice. 

I'll go ahead and make it so we don't set a new timer if one already exists, which makes sense because it's not like we'd want to extend the timer if it's already running.